posts/react-compound-components/2021-5-25-React-compound-compnents.md

```diff
@@ -0,0 +1,250 @@
+---
+templateKey: "blog-post"
+title: "React compound components (Hooks + typescript)"
+date: 2021-05-25
+featuredpost: false
+description: >-
+  This article talks about what are compound components and how to build an accordion component using the concept of compound components.
+keywords:
+- reactjs
+- Compound-components
+- Hooks
+- Typescript
+link: /react-compound-components
+category:
+- Tutorial
+author: Ashwin kumar
+tags:
+- react js
+- reactjs
+- typescript
+- compound components
+- build accordion component using compound components
+---
+
+**Compound Components** are an advanced and great pattern which you can use to create the meta component. It gives your components more flexibility. It is little difficult to understand the concept initially, But believe me, if you get a hold of it you will love building the components using compound components pattern
+
+## What are compound components?
+
+The compound components are a set of two or more components that work together to accomplish a specific task. The set of components will share an implicit state to communicate between them.
+
+> Think of compound components like the select and option elements in HTML. Apart they don’t do too much, but together they allow you to create the complete experience. — Kent C. Dodds
+
+```jsx
+<select>
+ <option>Option1</option>
+ <option>Option2</option>
+ <option>Option3</option>
+ <option>Option4</option>
+</select>
+```
+
+In the above example, when you click on an option in the select component, select knows which option you clicked. The select and the option share the state between them and update the selected option state on their own, we don’t need to explicitly configure them.
+
+## Compound Components in Action
+
+In this post, we’ll be building an Accordion component using compound components.
+
+You can check out the [Code sandbox link](https://codesandbox.io/s/interesting-wildflower-wj3iy) for the final demo of the accordion component.
```

Comment:
Instead of linking the codesandbox out, embed the codesandbox like in this page https://codebrahma.com/our-process/